src/browser/atom-application.coffee

```diff
@@ -369,6 +369,7 @@ class AtomApplication
   # Get the dimensions for opening a new window by cascading as appropriate to
   # the platform.
   getDimensionsForNewWindow: ->
+    return if (@focusedWindow() ? @lastFocusedWindow)?.isMaximized() and process.platform isnt 'darwin'
```

Comment:
I don't think we need to bother conditionalizing this since that's the right behavior for OS X, too.